dbmongo/js/reduce.algo2/delais.ts

```diff
@@ -0,0 +1,73 @@
+// Object golang défini dans dbmongo/lib/urssaf/delai.go
+export type Delai = {
+  numero_compte: string
+  numero_contentieux: string
+  date_creation: Date
+  date_echeance: Date
+  duree_delai: number // nombre de jours entre date_creation et date_echeance
+  denomination: string
+  indic_6m: string
+  annee_creation: number
+  montant_echeancier: number // exprimé en euros
+  stade: string
+  action: string
+}
+
+export type DelaiMap = { [key: string]: Delai }
+
+export function delais(v: { delai: DelaiMap }, output_indexed: object): void {
+  "use strict"
+  Object.keys(v.delai).map(function (hash) {
+    const delai = v.delai[hash]
+    // On arrondit les dates au premier jour du mois.
+    const date_creation = new Date(
+      Date.UTC(
+        delai.date_creation.getUTCFullYear(),
+        delai.date_creation.getUTCMonth(),
+        1,
+        0,
+        0,
+        0,
+        0
+      )
+    )
+    const date_echeance = new Date(
+      Date.UTC(
+        delai.date_echeance.getUTCFullYear(),
+        delai.date_echeance.getUTCMonth(),
+        1,
+        0,
+        0,
+        0,
+        0
+      )
+    )
+    // Création d'un tableau de timestamps à raison de 1 par mois.
+    const pastYearTimes = globalThis.f // variable globale f déclarée dans globals.ts
```

Comment:
Est-ce qu'on ne casse pas l'exécution du map-reduce ? Ou alors il faut mettre "globalThis.f" dans la variable "scope" injectée par golang ? (ça serait un peu hacky)

Reply:
Good catch! 👏
Effectivement:
> It’s also important to note that TypeScript doesn’t transform references to globalThis when compiling to older versions of ECMAScript. As such, unless you’re targeting evergreen browsers (which already support globalThis), you may want to use an appropriate polyfill instead.
(source: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis)
J'avoue que je n'avais pas pensé à ça. À moins que l'interpréteur JS de MongoDB colle très fidèlement aux dernières fonctionnalités du langage JavaScript/ECMAScript, je pense que tu as raison. => Je vais y réflechir, mais n'hésite pas si tu as des idées !

Reply:
Afin d'éviter d'avoir à injecter `globalThis` depuis golang, j'ai promu `f` en véritable variable globale.

L'avantage, c'est que ça devrait fonctionner comme d'habitude.

L'inconvénient (à mon avis), c'est que, même depuis nos fichiers TypeScript, on continue de référencer cette variable `f` qui ne semble venir de nulle part. Ceci dit, c'est pas très grave, à condition qu'on n'abuse pas trop des variables globale. (ce qui était justement l'esprit de la PR #40).